Source/santad/Logs/EndpointSecurity/Serializers/Utilities.mm

```diff
@@ -125,14 +124,44 @@ const mach_port_t GetDefaultIOKitCommsPort() {
   IORegistryEntryCreateCFProperties(device, &device_properties, kCFAllocatorDefault, kNilOptions);
   NSDictionary *properties = CFBridgingRelease(device_properties);
   IOObjectRelease(device);
+  return properties;
+}
+
+NSString *DiskImageForDevice(NSString *devPath) {
+  NSString *result;
+
+  // First try getting the image-path property of the parent path of
+  // the given device path. Since at least macOS 13, this mainly works
+  // for `hdiutl attach` only.
+  NSDictionary *properties = PropertiesForDevice([devPath stringByDeletingLastPathComponent]);
 
   if (properties[@"image-path"]) {
-    NSString *result = [[NSString alloc] initWithData:properties[@"image-path"]
-                                             encoding:NSUTF8StringEncoding];
-    return [result stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+    result = [[NSString alloc] initWithData:properties[@"image-path"]
+                                   encoding:NSUTF8StringEncoding];
   } else {
-    return nil;
+    // As a fallback, lookup properties of the full given path then look
```

Comment:
Given that this works for both mount routes on all recent macOSes, it would be better to do _this_ method first and fallback to the old way.

Reply:
My hesitation here is that while I think this is probably a true statement, I don't really know for sure... With that said, I could find and boot some old macOS images and see what things look like.

Reply:
That's why I didn't flag this earlier as well. The change that Matt made guarantees that we at least have the same behavior we've always had. 

Without stronger testing / a better understood root cause I'd be reluctant to change this to the default.

Reply:
So fwiw, I tested back to 13.6 and the new methods work the same way as they do in 15.3.1. Given we'll have some time to bake the changes before the next release, I went ahead and switched up the order. The nice thing about the new order is that we likely only ever do one fetch/copy of properties, where if we checked things the old way first we would often require two fetch/copies.